# HG changeset patch
# User Rik <rik@octave.org>
# Date 1346788466 25200
#      Tue Sep 04 12:54:26 2012 -0700
# Node ID 8639a450a89e298b1e8fd4ba2536c777cef8eba8
# Parent  ae9079bbc62733cb67c5509edf4976dc818d1353
build: Test for QT setPlaceholderText function before using it.

* webinfo.cc: build: Test for QT setPlaceholderText function before using it.

diff --git a/libgui/src/qtinfo/webinfo.cc b/libgui/src/qtinfo/webinfo.cc
--- a/libgui/src/qtinfo/webinfo.cc
+++ b/libgui/src/qtinfo/webinfo.cc
@@ -65,17 +65,20 @@ webinfo::webinfo (QWidget *parent)
 
   _stacked_widget = new QStackedWidget (this);
   layout->addWidget (_stacked_widget);
 
   hboxLayout = new QHBoxLayout ();
   layout->addLayout (hboxLayout);
 
   _search_line_edit = new QLineEdit(this);
+  // FIXME: maybe put in a better test in configure.ac
+#if QT_VERSION >= 0x040700
   _search_line_edit->setPlaceholderText (tr ("Type here and press \'Return\' to search"));
+#endif
   hboxLayout->addWidget (_search_line_edit);
 
   _search_check_box = new QCheckBox (tr ("Global search"));
   hboxLayout->addWidget (_search_check_box);
 
   connect (_close_tab_button, SIGNAL (clicked ()), this, SLOT (close_tab ()));
   connect (_tab_bar, SIGNAL (currentChanged (int)), this, SLOT (current_tab_changed (int)));
   connect (_zoom_in_button, SIGNAL (clicked ()), this, SLOT (zoom_in ()));
